### PR TITLE
Add tag on DM Annihilation spectral model

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request solves #4056  by simply adding a `tag` on `DarkMatterAnnihilationSpectralModel` with short name `dm-annihilation`.
Note that DM spatial models are not usable in `SkyModel` as they are not regular `SpatialModel`.

**Dear reviewer**

### CODE REVIEW

**Comment by adonath** [comment]
Thanks @registerrier! My main question is whether adding the tag really solves the problem? The `DarkMatterAnnihilationSpectralModel` as much more associated information , which needs to be serialized with a custom `.to_dict()` methods. Including `k, z, mass, channel`...

**Comment by registerrier** [comment]
> Thanks @registerrier! My main question is whether adding the tag really solves the problem? The `DarkMatterAnnihilationSpectralModel` as much more associated information , which needs to be serialized with a custom `.to_dict()` methods. Including `k, z, mass, channel`...

This is a good point. Technically, some of these quantities could be `Parameter` objects. But not e.g. `channel`.  How to serialize non `Parameter` attributes is not clear. 
Maybe this is a similar case to `NaimaSpectralModel` which has no yaml serialization for now.

**Comment by adonath** [comment]
@registerrier Yes, maybe let's just raise a meaningful error for now?

**Comment by registerrier** [comment]
I have included additional attributes in the out dictionary. It actually works and allows exporting DMmodels to yaml.

Obviously we will need to look at what should be a `Parameter` and what is a simple model attribute. But that could be left for later as the DM models will have to be better included in the `Model` framework (or they will have to be removed if those models are found to be useless).

Opinions @adonath ?

**Comment by adonath** [comment]
Thanks @registerrier, great to see it works with limited effort. I'll make a more thorough review later...

File: gammapy/astro/darkmatter/spectra.py [hunk 1]
```diff
@@ -200,3 +201,27 @@ def evaluate(self, energy, scale):
...
+        data = data["spectral"]
+
+        channel = data["channel"]
+        mass = u.Quantity(data["mass"])
+        jfactor = u.Quantity(data["jfactor"])
+        z = data["z"]
+        k = data["k"]
+
+        scale = [p["value"] for p in data["parameters"] if p["name"] == "scale"][0]
+        return cls(mass, channel, scale=scale, jfactor=jfactor, z=z, k=k)
```
Comment (adonath) [inline gammapy/astro/darkmatter/spectra.py:225]:
```suggestion
    def from_dict(cls, data):
        """Create spectral model from dict
        
        Parameters
        ----------
        data : dict
            DIct with model data
        
        Returns
        -------
        model : `DarkMatterAnnihilationSpectralModel`
            Dark matter annihilation spectral model
        """
        data = data["spectral"]
        scale = [p["value"] for p in data.pop("parameters") if p["name"] == "scale"][0]
        return cls(scale=scale, **data)
```

File: gammapy/astro/darkmatter/spectra.py [hunk 2]
```diff
@@ -177,13 +177,14 @@ class DarkMatterAnnihilationSpectralModel(SpectralModel):
         interp="log",
         is_norm=True,
     )
+    tag = ["DarkMatterAnnihilationSpectralModel", "dm-annihilation"]
 
     def __init__(self, mass, channel, scale=scale.quantity, jfactor=1, z=0, k=2):
         self.k = k
         self.z = z
-        self.mass = mass
+        self.mass = u.Quantity(mass)
```
Comment (adonath) [inline gammapy/astro/darkmatter/spectra.py:185]: Maybe add a unit test to establish the quantity here...

**Review by adonath — COMMENTED** [review]
Thanks @registerrier, I have left one minor comment and one inline suggestion...

**Comment by adonath** [comment]
What is the status here @registerrier ?

File: gammapy/astro/darkmatter/spectra.py [hunk 3]
```diff
@@ -200,3 +201,34 @@ def evaluate(self, energy, scale):
...
+        data : dict
+            Dict with model data
+
+        Returns
+        -------
+        model : `DarkMatterAnnihilationSpectralModel`
+            Dark matter annihilation spectral model
+        """
+        data = data["spectral"]
+        type = data.pop("type")
```
Comment (adonath) [inline gammapy/astro/darkmatter/spectra.py:229]:
```suggestion
        data.pop("type")
```

**Review by adonath — APPROVED** [review]
Thanks @registerrier, no further comments from my side.

**Pull request merged**